Automattic/happychat-client
# Add support for npm package

**Built on top of https://github.com/Automattic/happychat-client/pull/183**

## Summary
This PR prepares `happychat-client` to be exposed as a npm package. Paths are resolved using `babel-plugin-module-resolver` and configuration json files are inlined using `babel-plugin-inline-json`.

## Testing
1. Switch calypso to `try/happychat-client` branch (https://github.com/Automattic/wp-calypso/pull/23630)
2. Link this package to wp-calypso, by following these steps
- run `npm link` in this folder
- run `npm link happychat-client` in calypso folder
3. Run `npm install` in this branch
4. Run `npm run dev:standalone` in this repo
5. Grab your oauth token from a wpcom request made by the client (http://localhost:9000) it is under `Authorization` header after `Bearer`
6. Add the token to calypso https://github.com/Automattic/wp-calypso/pull/23630/files#diff-56de7193249f7b0107d6033c1a7737acR22
7. Run `npm dev:npm`
8. Access http://calypso.localhost:3000/me/chat
9. It should load the happychat-client iframe
